### PR TITLE
Homework12: Aksana Batyrova 

### PR DESCRIPTION
**Pull request by AksanaBatyrova**
HW 12 - OOP pt. 1

### CODE REVIEW

File: homework12/task1_library.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    def __init__(self, name, taken_book, reserved_book):
+        self.name = name
+        self.taken_book = taken_book
+        self.reserved_book = reserved_book
+
+    def reserved_by(self, name):
+        """This method return the name of book owner"""
+        return name
+
+    def take_book(self, cls):
```
Comment (Stml89) [inline homework12/task1_library.py:63]: `cls` is used with classmethod decorator, `self` is used in the class
https://www.geeksforgeeks.org/class-method-vs-static-method-python/

File: homework12/task1_library.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    Methods:
+    take_book(): assigns the book to user
+    reserve_book(): reserves book for user
+    return_book(): unassign the book from user
+    """
+
+    def __init__(self, name, taken_book, reserved_book):
+        self.name = name
+        self.taken_book = taken_book
+        self.reserved_book = reserved_book
```
Comment (Stml89) [inline homework12/task1_library.py:57]:
```suggestion
        self.taken_book = None
        self.reserved_book = None
```

File: homework12/task1_library.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        self.taken_book = None
+        cls.is_reserved = False
+        return f"You've returned '{cls.name}' by {cls.author}"
+
+
+allan_poe_raven = Book("The Raven", "Edgar Allan Poe", 6, 9788506007914)
+johann_goethe_faust = Book("Faust, a Tragedy", "Johann Wolfgang von Goethe",
+                           165, 1503262146)
+user1 = User("John", None, None)
+user2 = User("Jane", None, None)
```
Comment (Stml89) [inline homework12/task1_library.py:91]:
```suggestion
user1 = User("John")
user2 = User("Jane")
```

File: homework12/task2_bank_deposit.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    term: deposit term
+
+    Methods:
+    deposit(): calculating how much bank client will receive when depositing at
+    compound interest
+    """
+
+    def deposit(self, n, r):
+        """This method is for calculating compound interest"""
+        return int(n*(1+0.1)**r)
```
Comment (Stml89) [inline homework12/task2_bank_deposit.py:48]: use `round` insted

File: homework12/task2_bank_deposit.py [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    Arguments:
+    amount: money deposited
+    term: deposit term
+
+    Methods:
+    deposit(): calculating how much bank client will receive when depositing at
+    compound interest
+    """
+
+    def deposit(self, n, r):
```
Comment (Stml89) [inline homework12/task2_bank_deposit.py:46]: If the user puts 1000, for 1 year, with 10% -> expected to get ~1104.71

File: homework12/task2_bank_deposit.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+# сумма вклада и его срок. Пользователь делает вклад в размере N рублей сроком
+# на R лет под 10% годовых (вклад с возможностью ежемесячной капитализации -
+# это означает, что проценты прибавляются к сумме вклада ежемесячно). Написать
+# класс Bank, метод deposit принимает аргументы N и R, и возвращает сумму,
+# которая будет на счету пользователя.
+
+# https://myfin.by/wiki/term/kapitalizaciya-procentov
+
+
+class Deposit:
```
Comment (Stml89) [inline homework12/task2_bank_deposit.py:14]: you are not using this class, if so pls remove it

**Review by Stml89 — CHANGES_REQUESTED** [review]

File: homework12/task1_library.py (hunk 3 above)
Comment (AksanaBatyrova) [reply homework12/task1_library.py]: fixed

File: homework12/task2_bank_deposit.py (hunk 4 above)
Comment (AksanaBatyrova) [reply homework12/task2_bank_deposit.py]: fixed

File: homework12/task2_bank_deposit.py (hunk 5 above)
Comment (AksanaBatyrova) [reply homework12/task2_bank_deposit.py:26]: fixed

File: homework12/task2_bank_deposit.py (hunk 6 above)
Comment (AksanaBatyrova) [reply homework12/task2_bank_deposit.py]: fixed

File: homework12/task1_library.py (hunk 1 above)
Comment (AksanaBatyrova) [reply homework12/task1_library.py]: fixed

File: homework12/task2_bank_deposit.py (hunk 5 above)
Comment (Stml89) [reply homework12/task2_bank_deposit.py:26]: The customer wouldn’t be happy, because if she puts 1000, for 1 year, with 10% -> expected to get ~1104.71

Comment (AksanaBatyrova) [reply homework12/task2_bank_deposit.py]: fixed

**Review by Stml89 — APPROVED** [review]

**Pull request merged**